### PR TITLE
some notes on the virtual machine

### PR DESCRIPTION
**Pull request by davidweichiang**
Added a file ELVM.md with some more detailed description of the virtual machine than in README.md, which might be useful for anyone wanting to write a new backend. I hope it is all correct!

### CODE REVIEW

File: ELVM.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+- src: immediate or register
+- dst: register
+- no borrow flag
+
+LOAD dst, src
+- copy address src to dst
+- src: immediate or register
+- dst: register
+
+STORE dst, src
```
Comment (shinh) [inline ELVM.md]: The direction is wrong here.

> STORE src, dst

I know this looks weird comparing to other operations, but I think this is a fairly common convention in RISC CPUs. This makes hardware implementation slightly easier, IIUC.

**Comment by shinh** [comment]
Nice, thanks! I've pushed a follow up patch to fix the minor issue in STORE, and added a couple of links from README.md to ELVM.md

https://github.com/shinh/elvm/commit/6b10a1b3f50444899db24392656467dc2b0b6aa2

File: ELVM.md (hunk 1 above)
Comment (davidweichiang) [inline ELVM.md]: Internally, the Inst object uses src for the destination address and dest for the source address, though. So shouldn't it still be STORE dst, src, but the description should read "store value in dst to memory location src"?

Comment (shinh) [inline ELVM.md]: Very true. Thanks for pointing this out. I think ideally it'd be nicer to correct the internal implementation, but it's somewhat involved as there're already a bunch of backends... For the meantime, I've added a note on the discrepancy: https://github.com/shinh/elvm/commit/34e1ebb8b21e509cf1cfd71d05592516b46ce360

**Pull request merged**